frontend/src/metabase/static-viz/lib/numbers.ts

```diff
@@ -67,5 +70,26 @@ export const formatNumber = (number: number, options?: NumberFormatOptions) => {
   return `${prefix}${formattedNumber}${suffix}`;
 };
 
+// Simple hack to handle small decimal numbers (0-1)
+function handleSmallNumberFormat<T>(value: number, options: T): T {
+  const hasAtLeastThreeDecimalPoints = Math.abs(value) < 0.01;
```

Comment:
From this [Slack thread](https://metaboat.slack.com/archives/C5XHN8GLW/p1663000412851289), the previous change was breaking some assumptions for [certain tests](https://github.com/metabase/metabase/actions/runs/3038567058), and I only want to address the fix for small numbers that have 3 decimal points and more since we format all numbers to have at most 2 decimal points by default if the desired number of decimal points is not specified via `decimals` option